Assets/Scripts/2D/StartGuiManagerScript.cs

```diff
@@ -86,6 +92,8 @@ void Start()
         LoadButton.interactable = HasSaveFilesToLoad();
     }
 
+    /// <summary>Called when the StartGuiManager script instance is being loaded.</summary>
+    /// <remarks>Used to initialize settings before the game starts.</remarks>
     void Awake()
```

Comment:
Classes that inherit from MonoBehaviour are classes that Unity can directly associated to objects in the Unity3D editor. These classes can "overload" certain MonoBehaviour methods in order to modify the object behavior in Unity. Some of these functions are OnEnable, OnDisable, OnDestroy, Awake, Start, and Update. What they have in common is that they do not use access modifiers ('public', 'private', etc).

So, I think adding specific descriptions to them might be misleading since the "Awake" method in one class might do a complete different thing that the "Awake" method in another, but from a general point of view, their purpose is exactly the same.

Just take into account when documenting functions that inherit from Monobehaviour. You can consult the Monobehaviour documentation to find which functions are common (https://docs.unity3d.com/ScriptReference/MonoBehaviour.html)